suite-native/receive/src/components/DeviceScreenContent.tsx

```diff
@@ -104,17 +110,31 @@ export const DeviceScreenContent = ({
     const deviceFont = useFont(fontSource, fontSize);
     const { translate } = useTranslate();
     const [visibleDots, setVisibleDots] = useState<number>(1);
+    const [visualState, setVisualState] = useState<DeviceScreenVisualState>({
+        isAddressShown: false,
+        isPaginationShown: false,
+        isLoading: false,
+    });
+
+    useEffect(() => {
```

Comment:
Redundant useEffect. You can simply do just:

```tsx
const visualState = {
  isLoading:  isAddressRevealed && !isConfirmOnTrezorReady && !isReceiveApproved,
  isAddrressShown:  isAddressRevealed && isConfirmOnTrezorReady,
  ...
}
```

Reply:
rewritten